drivers/gnss/quectel_lcx6g.c

```diff
@@ -0,0 +1,757 @@
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	data->dynamic_match.match_size = ret;
+
+	ret = modem_chat_run_script(&data->chat, &data->dynamic_script);
+	k_spin_unlock(&data->lock, key);
+	return ret;
+}
+
+static void quectel_lcx6g_get_search_mode_callback(struct modem_chat *chat, char **argv,
+						   uint16_t argc, void *user_data)
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)user_data;
+
+	if (argc != 8) {
+		return;
+	}
+
+	data->enabled_systems_response = ('1' == argv[1][0]) ? GNSS_SYSTEM_GPS : 0;
+	data->enabled_systems_response |= ('1' == argv[2][0]) ? GNSS_SYSTEM_GLONASS : 0;
+	data->enabled_systems_response |= ('1' == argv[3][0]) ? GNSS_SYSTEM_GALILEO : 0;
+	data->enabled_systems_response |= ('1' == argv[4][0]) ? GNSS_SYSTEM_BEIDOU : 0;
+	data->enabled_systems_response |= ('1' == argv[5][0]) ? GNSS_SYSTEM_QZSS : 0;
+}
+
+static void quectel_lcx6g_get_sbas_status_callback(struct modem_chat *chat, char **argv,
+						   uint16_t argc, void *user_data)
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)user_data;
+
+	if (argc != 3) {
+		return;
+	}
+
+	data->enabled_systems_response |= ('1' == argv[1][0]) ? GNSS_SYSTEM_SBAS : 0;
+}
+
+
+static int quectel_lcx6g_get_enabled_systems(const struct device *dev, gnss_systems_t *systems)
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)dev->data;
+	k_spinlock_key_t key;
+	int ret;
+
+	key = k_spin_lock(&data->lock);
+
+	ret = gnss_nmea0183_snprintk(data->dynamic_request_buf, sizeof(data->dynamic_request_buf),
+				     "PAIR067");
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	data->dynamic_script_chat.request_size = ret;
+
+	strncpy(data->dynamic_match_buf, "$PAIR067,", sizeof(data->dynamic_match_buf));
+	data->dynamic_match.match_size = sizeof("$PAIR067,") - 1;
+	data->dynamic_match.callback = quectel_lcx6g_get_search_mode_callback;
+
+	ret = modem_chat_run_script(&data->chat, &data->dynamic_script);
+	data->dynamic_match.callback = NULL;
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	ret = gnss_nmea0183_snprintk(data->dynamic_request_buf, sizeof(data->dynamic_request_buf),
+				     "PAIR411");
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	data->dynamic_script_chat.request_size = ret;
+
+	strncpy(data->dynamic_match_buf, "$PAIR411,", sizeof(data->dynamic_match_buf));
+	data->dynamic_match.match_size = sizeof("$PAIR411,") - 1;
+	data->dynamic_match.callback = quectel_lcx6g_get_sbas_status_callback;
+
+	ret = modem_chat_run_script(&data->chat, &data->dynamic_script);
+	data->dynamic_match.callback = NULL;
+	if (ret < 0) {
+		k_spin_unlock(&data->lock, key);
+		return ret;
+	}
+
+	*systems = data->enabled_systems_response;
+	return 0;
+}
+
+static int quectel_lcx6g_get_supported_systems(const struct device *dev, gnss_systems_t *systems)
+{
+	(*systems) = (GNSS_SYSTEM_GPS | GNSS_SYSTEM_GLONASS | GNSS_SYSTEM_GALILEO |
```

Comment:
do you need the parenthesis? `*systems = (...)`

Reply:
Nope, removed :)